pkg/util/conditions/conditions.go

```diff
@@ -0,0 +1,74 @@
+package conditions
+
+import (
+	"fmt"
+
+	"github.com/ghodss/yaml"
+	"github.com/golang/glog"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+// GetNodeCondition returns node condition by type
+func GetNodeCondition(node *corev1.Node, conditionType corev1.NodeConditionType) *corev1.NodeCondition {
+	for _, cond := range node.Status.Conditions {
+		if cond.Type == conditionType {
+			return &cond
+		}
+	}
+	return nil
+}
+
+// UnhealthyConditions contains a list of UnhealthyCondition
+type UnhealthyConditions struct {
+	Items []UnhealthyCondition `json:"items"`
+}
+
+// UnhealthyCondition is the representation of unhealthy conditions under the config map
+type UnhealthyCondition struct {
+	Name    corev1.NodeConditionType `json:"name"`
+	Status  corev1.ConditionStatus   `json:"status"`
+	Timeout string                   `json:"timeout"`
+}
+
+// GetNodeUnhealthyConditions returns node unhealthy conditions
+func GetNodeUnhealthyConditions(node *corev1.Node, cmUnealthyConditions *corev1.ConfigMap) ([]UnhealthyCondition, error) {
+	data, ok := cmUnealthyConditions.Data["conditions"]
+	if !ok {
+		return nil, fmt.Errorf("can not find \"conditions\" under the configmap")
+	}
+
+	var unealthyConditions UnhealthyConditions
+	err := yaml.Unmarshal([]byte(data), &unealthyConditions)
+	if err != nil {
+		glog.Errorf("failed to umarshal: %v", err)
+		return nil, err
+	}
+
+	conditions := []UnhealthyCondition{}
+	for _, c := range unealthyConditions.Items {
+		cond := GetNodeCondition(node, c.Name)
+		if cond != nil && cond.Status == c.Status {
+			conditions = append(conditions, c)
+		}
+	}
+	return conditions, nil
+}
+
+// CreateDummyUnhealthyConditionsConfigMap creates dummy config map with default unhealth conditions
+func CreateDummyUnhealthyConditionsConfigMap() (*corev1.ConfigMap, error) {
+	unhealthyConditions := &UnhealthyConditions{
+		Items: []UnhealthyCondition{
+			{
+				Name:    "Ready",
+				Status:  "Unknown",
```

Comment:
why unknown? what about false?

Reply:
I added False condition as well, but again I do not sure when it appears at all.